site/source/docs/executing_scripts.md

```diff
@@ -0,0 +1,115 @@
+title: Executing Scripts
+layout: docs
+---
+
+There are various features in Embark that help you making the deployment of your DApps and Smart Contracts as smooth as possible. Next to general [Smart Contract Configurations](contracts_configuration.html), [Deployment Hooks](contracts_configuration.html#Deployment-hooks) and [Cockpit](cockpit_introduction.html), there's the possibility to run (migration) scripts as well.
+
+In this guide we'll explore why scripts are useful and how they can be run.
+
+## Why scripts?
+
+Given that Embark supports [afterDeploy](contracts_configuration.html#afterDeploy-hook) hooks that make it extremely easy to perform custom operations after all of your Smart Contracts have been deployed, you might wonder when and where scripts can be useful.
+
+It's important to note that `afterDeploy` hooks are executed every time all Smart Contracts have been deployed. Often there are cases where running a (migration) script manually is what you really need.
+
+Scripts let you exactly that as they can be run at any time, regardless of what your app's current deployment status is.
+
+## What's a script?
+
+A script is really just a file with an exported function that has special dependencies injected into it. Here's what it could look like:
+
+```
+modules.exports = async ({ contracts, web3, logger}) => {
+  ...
+};
+```
+
+The injected parameters are:
+
+- `contracts` - A map object containing all of your Smart Contracts as Embark Smart Contract instances.
+- `web3` - A web3 instances to give you access to things like accounts.
+- `logger` - Embark's custom logger.
+
+Scripts can be located anywhere on your machine but should most likely live inside your project's file tree in a dedicated folder.
+
+## Running scripts
+
+To run a script, use the CLI `exec` command and specify an environment as well as the script to be executed:
+
+```
+$ embark exec development scripts/001.js
+```
+
+The command above will execute the function in `scripts/001.js` and ensures that Smart Contracts are deployed in the `development` environment.
+
+If you have multiple scripts that should run in order it's also possible to specify the directory in which they live in:
+
+```
+$ embark exec development scripts
+```
+
+Embark will then find all script files inside the specified directory (in this case `scripts`) and then run them one by one. If any of the scripts fails by emitting an error, Embark will abort the execution. Scripts are executed in sequence which means, all following scripts won't be executed in case of an error.
```

Comment:
```suggestion
Embark will then find all script files inside the specified directory (in this case `scripts`) and then run them one by one. If any of the scripts fails by emitting an error, Embark will abort the execution. Scripts are executed in sequence, which means all following scripts won't be executed in case of an error.
```
Not sure about this one ^